File: app/src/main/java/com/proxy/shadowsocksr/ui/DialogManager.java
package com.proxy.shadowsocksr.ui;

import android.app.ProgressDialog;
import android.content.Context;

public final class DialogManager
{
    private DialogManager()
    {
    }

    private static DialogManager manager;

    public static DialogManager getInstance()
    {
        if (manager == null)
        {
            manager = new DialogManager();
        }
        return manager;
    }

    private ProgressDialog pd;

    public void showTipDialog(Context cxt, int resid)
    {
        pd = new ProgressDialog(cxt);
        pd.setMessage(cxt.getString(resid));
        pd.setIndeterminate(true);
        pd.setCancelable(false);
        pd.show();
    }

    public void dismissTipDialog()
    {
        if (pd != null)
        {
            pd.dismiss();
            manager = null;
        }
    }
}

File: app/src/main/java/com/proxy/shadowsocksr/ui/DialogManager.kt
package com.proxy.shadowsocksr.ui

import android.app.ProgressDialog
import android.content.Context

object DialogManager
{
    private var pd: ProgressDialog? = null

    fun showTipDialog(cxt: Context, resid: Int)
    {
        pd = ProgressDialog(cxt)
        pd!!.setMessage(cxt.getString(resid))
        pd!!.isIndeterminate = true
        pd!!.setCancelable(false)
        pd!!.show()
    }

    fun dismissTipDialog()
    {
        if (pd != null)
        {
            pd!!.dismiss()
            pd = null
        }
    }
}
